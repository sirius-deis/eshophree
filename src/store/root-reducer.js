import { combineReducers } from "redux";

import categoryReducer from "./category/category.reducer";
import userReducer from "./user/user.reducer";
import cartReducer from "./cart/cart.reducer";

const rootReducer = combineReducers({
    category: categoryReducer,
    user: userReducer,
    cart: cartReducer,
});

export default rootReducer;
