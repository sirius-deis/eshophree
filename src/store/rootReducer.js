import { combineReducers } from 'redux';

import userReducer from './user/user.reducer';
import categoryReducer from './category/category.reducer';

const rootReducer = combineReducers({
  user: userReducer,
  category: categoryReducer,
});

export default rootReducer;
