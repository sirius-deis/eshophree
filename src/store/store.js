import { createStore, applyMiddleware } from 'redux';
import { persistReducer, persistStore } from 'redux-persist';
import storage from 'redux-persist/lib/storage';
import logger from 'redux-logger';
import thunk from 'redux-thunk';

import rootReducer from './rootReducer';

const persistConfig = {
  key: 'root',
  storage,
};

const persistedReducer = persistReducer(persistConfig, rootReducer);

const middlewares = [thunk];

if (process.env.NODE_ENV === 'development') {
  middlewares.push(logger);
}

export const store = createStore(persistedReducer, applyMiddleware(...middlewares));

export const persistor = persistStore(store);
