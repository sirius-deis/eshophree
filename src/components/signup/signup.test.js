import { render, screen, fireEvent, act } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../../store/store';
import SignUp from './signup';

global.fetch = jest.fn(() =>
  Promise.resolve({
    json: () =>
      Promise.resolve({
        message: 'Done',
      }),
  }),
);

describe('SignUp component', () => {
  it('should match snapshot', () => {
    const { container } = render(
      <Provider store={store}>
        <MemoryRouter>
          <SignUp />
        </MemoryRouter>
      </Provider>,
    );
    expect(container).toMatchSnapshot();
  });
  it('should fire submit event', async () => {
    render(
      <Provider store={store}>
        <MemoryRouter>
          <SignUp />
        </MemoryRouter>
      </Provider>,
    );
    const emailInput = screen.getByLabelText('Email *');
    fireEvent.change(emailInput, {
      target: {
        value: 'test@test.com',
      },
    });
    const passwordInput = screen.getByLabelText('Password *');
    fireEvent.change(passwordInput, {
      target: {
        value: 'password',
      },
    });
    const passwordConfirm = screen.getByLabelText('Password confirm *');
    fireEvent.change(passwordInput, {
      target: {
        value: 'password',
      },
    });
    const form = screen.getByRole('form', { hidden: true });
    // eslint-disable-next-line testing-library/no-unnecessary-act
    await act(() => {
      fireEvent.submit(form);
    });
    expect(emailInput.value).toBe('');
    expect(passwordInput.value).toBe('');
    expect(passwordConfirm.value).toBe('');
  });
});
