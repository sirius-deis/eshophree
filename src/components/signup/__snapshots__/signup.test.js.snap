// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp component should match snapshot 1`] = `
<div>
  <form
    aria-label="form"
    class="sc-bgqQcB gmCWKN"
  >
    <h2
      class="sc-gTRrQi qTUwf"
    >
      Sign up
    </h2>
    <div
      class="sc-iVCKna dqAret"
    >
      <label
        class="sc-gxYJeL dLwkrg"
      >
        Email
         *
        <input
          class="sc-kPTPQs ebvaXq"
          name="email"
          placeholder="Enter your email"
          required=""
          type="email"
        />
      </label>
    </div>
    <div
      class="sc-iVCKna dqAret"
    >
      <label
        class="sc-gxYJeL dLwkrg"
      >
        Password
         *
        <input
          class="sc-kPTPQs ebvaXq"
          minlength="8"
          name="password"
          placeholder="Enter your password"
          required=""
          type="password"
        />
      </label>
    </div>
    <div
      class="sc-iVCKna dqAret"
    >
      <label
        class="sc-gxYJeL dLwkrg"
      >
        Password confirm
         *
        <input
          class="sc-kPTPQs ebvaXq"
          minlength="8"
          name="confirm"
          placeholder="Confirm your password"
          required=""
          type="password"
        />
      </label>
    </div>
    <div
      style="display: flex; justify-content: space-between; font-size: 1.3rem;"
    >
      <p>
        Have account?
         
        <a
          href="/login"
          style="text-decoration: underline; padding-left: 0.5rem;"
        >
          Sign in
        </a>
      </p>
    </div>
    <button
      class="sc-ispOId sc-kuWgmH dHSeuY jjKghz"
    >
      Sign up
    </button>
    <div
      style="display: flex; justify-content: center; align-items: center; gap: 1rem;"
    >
      <div
        class="sc-ewnqHT bVIEaX"
      />
      <p
        style="font-size: 1.3rem;"
      >
        Or sign up with
      </p>
      <div
        class="sc-ewnqHT bVIEaX"
      />
    </div>
    <div
      style="display: flex; justify-content: space-between; align-items: center; gap: 2rem;"
    >
      <button
        class="sc-ispOId sc-kuWgmH hIhUdR jjKghz"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 320 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M279.14 288l14.22-92.66h-88.91v-60.13c0-25.35 12.42-50.06 52.24-50.06h40.42V6.26S260.43 0 225.36 0c-73.22 0-121.08 44.38-121.08 124.72v70.62H22.89V288h81.39v224h100.17V288z"
          />
        </svg>
         Facebook
      </button>
      <button
        class="sc-ispOId sc-kuWgmH hIhUdR jjKghz"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 488 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M488 261.8C488 403.3 391.1 504 248 504 110.8 504 0 393.2 0 256S110.8 8 248 8c66.8 0 123 24.5 166.3 64.9l-67.5 64.9C258.5 52.6 94.3 116.6 94.3 256c0 86.5 69.1 156.6 153.7 156.6 98.2 0 135-70.4 140.8-106.9H248v-85.3h236.1c2.3 12.7 3.9 24.9 3.9 41.4z"
          />
        </svg>
         Google
      </button>
    </div>
  </form>
</div>
`;
